src/LogicBlock/SMTLibLogicModel.cpp

```diff
@@ -0,0 +1,131 @@
+
+#include "LogicBlock/SMTLibLogicModel.hpp"
+
+#include <regex>
+
+std::string smtliblogic::SMTLibLogicModel::getTermString(const LogicTerm& a) {
+    std::stringstream ss(modelString);
+    std::string       line;
+
+    const auto  regex = std::regex("\\(define-fun (.*) \\(\\) (.*) (.*)\\)");
+    std::smatch m;
+    while (std::getline(ss, line)) {
+        if (std::regex_search(line, m, regex)) {
+            if (m.size() == 4) {
+                if (m[1] == a.getName()) {
+                    return line;
+                }
+            }
+        }
+    }
+    return {};
+}
+std::string smtliblogic::SMTLibLogicModel::getTermValue(const logicbase::LogicTerm& a) {
+    std::string const termString = getTermString(a);
+
+    const auto  regex = std::regex(R"(\(define-fun (.*) \(\) (.*) (.*)\))");
```

Comment:
this is the same Regex as above. Since C++ regex are expensive to construct and the regex here and above are constructed on every function call, consider making the regex a protected member of the class itself. That should speed things up quite a bit.